\chapter{Introduction}
\label{intro}

Recent advances in {DNA} sequencing, specifically in next-generation sequencing~(NGS), revolutionised the field of genomics, making possible the generation of large amounts of sequencing data very rapidly and at substantially low cost~\cite{mardis2017dna}. This new technology also brought with it several challenges, namely in what concerns the analysis, storage, and transmission of the generated sequences~\cite{brouwer2016current,liu2012comparison}. As a consequence, several specialised tools were developed throughout the years in order to deal with these challenges.

Firstly, the storage of the raw data generated by NGS experiments is possible by using several file formats, the FASTQ and FASTA are the most commonly used~\cite{zhang2016overview}. %Therefore, its analysis and manipulation are crucial~\cite{buermans2014next}.
FASTQ is an extension of the FASTA format, that besides the nucleotide sequence, also stores associated per base quality score and it is considered the standard format for sequencing data storage and exchange~\cite{cock2009sanger}.

Regarding the analysis and manipulation of these sequencing data files many software applications emerged, including \texttt{fqtools}~\cite{droop2016fqtools},
\texttt{FASTX-Toolkit}~\cite{gordon2010fastx},
\texttt{GALAXY}~\cite{afgan2018galaxy}, \texttt{GATK}~\cite{depristo2011framework},
\texttt{MEGA}~\cite{kumar2016mega7}, \texttt{SeqKit}~\cite{shen2016seqkit}, among others. \texttt{Fqtools} is a suite of tools to view, manipulate and summarise FASTQ data. This software also identifies invalid FASTQ files~\cite{droop2016fqtools}. \texttt{GALAXY}, in its turn, is an open, web-based scientific platform for analysing genomic data~\cite{goecks2010galaxy}. This platform integrates several specialised sets of tools, e.g. for manipulating FASTQ files~\cite{blankenberg2010manipulation}.
\texttt{FASTX-Toolkit} is a collection of command-line tools to process FASTA and FASTQ files. This toolkit is available in two forms: as a command-line, or integrated into the web-based platform GALAXY~\cite{gordon2010fastx}. \texttt{SeqKit} is another toolkit used to process FASTA and FASTQ files and is available for all major operating systems~\cite{shen2016seqkit}. The Genome Analysis Toolkit~(\texttt{GATK}) was designed as a structured programming framework to simplify the development of analysis tools. However, nowadays, it is a suite of tools focused on variant discovering and genotyping~\cite{van2013fastq}. More towards the evolutionary perspectives, Molecular Evolutionary Genetics Analysis~(\texttt{MEGA}) software provides tools to analyse DNA and protein sequences statistically~\cite{tamura2011mega5}. Several of these frameworks lack on variety, namely the ability to perform multiple tasks using only one toolkit.

Compression is another important aspect when dealing with high-throughput sequencing data, as it reduces storage space and accelerates data transmission. A survey on DNA compressors and amino acid sequence compression can be found in~\cite{hosseini2016survey}. Currently, the DNA sequence compressors HiRGC~\cite{liu2017high}, iDoComp~\cite{ochoa2014idocomp}, GeCo~\cite{pratas2016efficient}, and GDC~\cite{deorowicz2015gdc} are considered to have the best performance~\cite{hernaez2019genomic}. Of these four approaches, GeCo is the only one that can be used for reference-free and reference-based compression. Furthermore, GeCo can be used as an analysis tool to determine absolute measures for many distance computations and local measures~\cite{pratas2016efficient}.

Amino acid sequences are known to be very hard to compress~\cite{nalbantoglu2010data}, however, Hosseini et al.~\cite{hosseini2019ac} recently developed AC, a state-of-the-art for lossless amino acid sequence compression. In~\cite{pratas2018compression} the authors compared the performance of AC, in terms of bit-rate, to several general-purpose lossless compressors and several protein compressors, using different proteomes. They concluded that in average AC provides the best bit-rates.

Another relevant subject is genomic data simulation. Read simulations tools are fundamental for the development, testing and evaluation of methods and computational tools~\cite{huang2011art,price2017simulome}. Despite the availability of a large number of real sequence reads, read simulation data is necessary due to the inability to know the ground truth of real data~\cite{baruzzo2017simulation}. Escalona \textit{et al.}~\cite{escalona2016comparison}, recently, reviewed 23 NGS simulation tools. XS~\cite{pratas2014xs}, a FASTQ read simulation tool, stands out in relation to the other 22 simulation tools because it is the only one that does not need a reference sequence. Furthermore, XS is the only open-source tool for simulation of FASTQ reads produced by the four most used sequencing machines, Roche-454, Illumina, ABI SOLiD and Ion Torrent. 

Although a large number of tools are available for analysing, compressing, and simulation, these tools are specialised in only a specific task. Besides, in many cases the output of one tool cannot be used directly as input for another tool, e.g. the output of a simulation tool cannot always be used directly as input for an analysis tool. Thus, unique software that includes several specialised tools is necessary.

In this document, we describe \texttt{GTO}, a complete toolkit for genomics and proteomics, namely for FASTQ, FASTA and SEQ formats, with many complementary tools. The toolkit is for Unix-based systems, built for ultra-fast computations. \texttt{GTO} supports pipes for easy integration with the sub-programs belonging to \texttt{GTO} as well as external tools. \texttt{GTO} works as \textit{LEGOs}, since it allows the construction of multiple pipelines with many combinations.

\texttt{GTO} includes tools for information display, randomisation, edition, conversion, extraction, search, calculation, compression, simulation and visualisation. \texttt{GTO} is prepared to deal with very large datasets, typically in the scale of Gigabytes or Terabytes (but not limited).
The complete toolkit is an optimised command-line version, using the prefix ``\texttt{gto}\_'' followed by the suffix with the respective name of the program. \texttt{GTO} is implemented in \texttt{C} language and it is available, under the MIT license, at:
\begin{lstlisting}
http://bioinformatics.ua.pt/gto
\end{lstlisting}

\section{Installation}
To install \texttt{GTO} through the GitHub repository:
\begin{lstlisting}
git clone https://github.com/cobilab/gto.git
cd gto/src/
make
\end{lstlisting}
Or by installing them directly using the Cobilab channel from Conda:
\begin{lstlisting}
conda install -c cobilab gto --yes
\end{lstlisting}


\section{Testing}
The examples provided in this document are available in the repository. Therefore, each example can be easily reproduced, which it will also test and validate each tool. To replicate those tests, it can be done in two different ways: 
\begin{itemize}
	\item Running one test for a specific tool:
\begin{lstlisting}
cd gto/tester/gto_{tool}
sh runExample.sh
\end{lstlisting}
\item Running the batch of tests for all the tools:
\begin{lstlisting}
cd gto/tester/
sh runAllTests.sh
\end{lstlisting}
\end{itemize}
Some of these tests require internet connection to download external files and it will create new files.

\section{Execution control}
The quality control in Unix/Linux pipelines using GTO's tools is made in three ways:  
\begin{enumerate}
\item Input verification: where the tools verify the format of the input file; 

\item  Stderr logs: Some execution errors are directly sent for the stderr channel. 

\item Scripting validation: In complex pipelines, the verification of all the tools in the pipeline were executed properly, it is used the PIPESTATUS variable, e.g.:  
\begin{lstlisting}
gto_fasta_rand_extra_chars < input.fa | gto_fasta_to_seq > output.seq 
echo "${PIPESTATUS[0]} ${PIPESTATUS[1]}" 
0 0 
\end{lstlisting}

\end{enumerate}


\section{License}

The license is \textbf{MIT}. In resume, it is a short and simple permissive license with conditions only requiring preservation of copyright and license notices. Licensed works, modifications, and larger works may be distributed under different terms and without source code.\\
\textbf{Permissions}:
\begin{itemize}
	\item commercial use;
	\item modification;
	\item distribution;
	\item private use.
\end{itemize}
\textbf{Limitations}:
\begin{itemize}
	\item liability;
	\item warranty.
\end{itemize}
\textbf{Conditions}:
\begin{itemize}
        \item License and copyright notice.
\end{itemize}
For details on the license, consult: \url{https://opensource.org/licenses/MIT}.
